### PR TITLE
Add knative install on ICP in the doc

### LINKED ISSUE #557: Adding a document for how to install Knative on IBM Cloud Private
/cc @zxdiscovery

### PR DESCRIPTION
**Pull request by zxDiscovery**
Fixes #557

## Proposed Changes

*  Add the a new file install/Knative-with-ICP.md
*  Update the install/README.md

### CODE REVIEW

File: install/Knative-with-ICP.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Knative Install on ICP
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply install/Knative-with-ICP.md]: +1, we should never use short name for some public document, please use `IBM Cloud Private` instead.

Also please remove `ICP` from this document.

File: install/Knative-with-ICP.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+Knative requires a ICP(IBM Cloud Private) cluster v2.1.0.3 or newer. 
+
+## Installing Istio
+
+Knative depends on Istio. Run the following to install Istio. (We are changing
+`LoadBalancer` to `NodePort` for the `istio-ingress` service).
+
+```shell
+curl -L https://raw.githubusercontent.com/knative/serving/v0.2.1/third_party/istio-1.0.2/istio.yaml \
```
Comment (gyliu513) [inline install/Knative-with-ICP.md]: Please use v0.2.2

File: install/Knative-with-ICP.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  ```shell
+  echo $(ICP cluster ip):$(kubectl get svc knative-ingressgateway --namespace istio-system --output 'jsonpath={.spec.ports[?(@.port==80)].nodePort}')
+  ```
+
+## Cleaning up
+
+Delete the Kubernetes cluster along with Knative, Istio, and any deployed apps:
+
+```shell
+kubectl delete
```
Comment (gyliu513) [inline install/Knative-with-ICP.md]: delete what?

File: install/Knative-with-ICP.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+ibm-privileged-psp          true      *                                                                                                                     RunAsAny   RunAsAny           RunAsAny    RunAsAny    false            *
+ibm-restricted-psp          false                                                                                                                           RunAsAny   MustRunAsNonRoot   MustRunAs   MustRunAs   false            configMap,emptyDir,projected,secret,downwardAPI,persistentVolumeClaim
+```
+
+Create a cluster role for the pod security policy resource. The resourceNames for this role must be the name of the pod security policy that was created previous. Here we use ``ibm-privileged-psp``.
+Create a YAML file for the cluster role.
+```shell
+vim knative-clusterrole.yaml
+```
+```shell
```
Comment (gyliu513) [inline install/Knative-with-ICP.md]: this is not `shell` but `yaml`, ditto for other YAML files.

**Comment by samodell** [comment]
@zxDiscovery  Thanks for your PR! Please address Averi and Guang's comments to start. We want to keep these install guides as similar as possible, and there are some global changes pending to the install guides in #565, so you'll want to copy the changes there once they're merged in. Adding a hold for that reason.

**Comment by zxDiscovery** [comment]
There are some mistake on this pull request. So close this and I will create a new one

**Pull request closed without merging**